### PR TITLE
Building c2hs

### PR DESCRIPTION
**Pull request by gdeest-da**
This PR builds `c2hs` with `Hazel` on Windows, which brings me just another step closer to building `gRPC-haskell`.

We had to patch `c2hs` to work around some issue with Cabal-style version macros in GHC 8.6. It looks like package versions are not properly sourced for core packages, which leads to `MIN_VERSION_base` not being defined.

Also, we pass `-XNoMonadFailDesugaring` in order to build the `c2hs` version within our Stackage snapshot with GHC 8.6.

### CODE REVIEW

File: 3rdparty/haskell/BUILD.c2hs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (aherrmann-da) [inline 3rdparty/haskell/BUILD.c2hs]:
```suggestion
```

File: 3rdparty/haskell/c2hs-package.bzl [hunk 2]
```diff
@@ -0,0 +1,522 @@
+package = (
```
Comment (aherrmann-da) [inline 3rdparty/haskell/c2hs-package.bzl]: I assume that file was generated using `cabal2bazel` and then modified manually?

Could you highlight where it was modified in a comment, so it's easy to see if we can go back to generated at some point in the future?

Comment (gdeest-da) [reply 3rdparty/haskell/c2hs-package.bzl]: Indeed. PTAL.

Comment (aherrmann-da) [reply 3rdparty/haskell/c2hs-package.bzl]: Thanks! LGTM

**Pull request merged**